compiler/executable/match_/planner/plan.rs

```diff
@@ -1987,13 +1991,18 @@ impl ConjunctionPlan<'_> {
 
 #[derive(Clone, Debug)]
 pub(super) struct DisjunctionPlanBuilder<'a> {
+    branch_ids: Vec<BranchID>,
     branches: Vec<ConjunctionPlanBuilder<'a>>,
     required_inputs: Vec<Variable>,
 }
 
 impl<'a> DisjunctionPlanBuilder<'a> {
-    fn new(branches: Vec<ConjunctionPlanBuilder<'a>>, required_inputs: Vec<Variable>) -> Self {
-        Self { branches, required_inputs }
+    fn new(
+        branch_ids: Vec<BranchID>,
+        branches: Vec<ConjunctionPlanBuilder<'a>>,
```

Comment:
Could also be maps IMO

Reply:
Best seems to be to introduce a new `struct Branch`, but it doesn't seem worth it at the moment.